diff --git a/snuba/cli/api.py b/snuba/cli/api.py
index ec6f29d3..00ff7a85 100644
--- a/snuba/cli/api.py
+++ b/snuba/cli/api.py
@@ -9,4 +9,4 @@ def api(*, debug: bool) -> None:
     from werkzeug.serving import WSGIRequestHandler
 
     WSGIRequestHandler.protocol_version = "HTTP/1.1"
-    application.run(port=settings.PORT, threaded=True, debug=debug)
+    application.run(port=settings.PORT, threaded=True, debug=debug, host=settings.HOST)
