diff --git a/py/setup.py b/py/setup.py
index 0cc3698..c793c35 100644
--- a/py/setup.py
+++ b/py/setup.py
@@ -83,6 +83,16 @@ def build_native(spec):
         rtld_flags=rtld_flags,
     )
 
+def find_native(spec):
+    rtld_flags = ["NOW"]
+    if sys.platform == "darwin":
+        rtld_flags.append("NODELETE")
+    spec.add_cffi_module(
+        module_path="symbolic._lowlevel",
+        dylib=lambda: "@nixBuildDylib@",
+        header_filename=lambda: "@nixBuildHeader@",
+        rtld_flags=rtld_flags,
+    )
 
 setup(
     name="symbolic",
@@ -98,6 +108,6 @@ setup(
     platforms="any",
     install_requires=["milksnake>=0.1.2",],
     setup_requires=["milksnake>=0.1.2",],
-    milksnake_tasks=[build_native,],
+    milksnake_tasks=[find_native,],
     cmdclass={"sdist": CustomSDist,},
 )
